backend/v3/api/org/v2/convert/org.go

```diff
@@ -0,0 +1,152 @@
+	toReturn := &v2_org.SearchQuery{}
+
+	switch assertedType := query.GetFilter().(type) {
+	case *v2beta_org.OrganizationSearchFilter_DomainFilter:
+		toReturn.Query = &v2_org.SearchQuery_DomainQuery{
+			DomainQuery: &v2_org.OrganizationDomainQuery{
+				Domain: assertedType.DomainFilter.GetDomain(),
+				Method: TextQueryMethodBetaToV2(assertedType.DomainFilter.GetMethod()),
+			},
+		}
+
+	case *v2beta_org.OrganizationSearchFilter_IdFilter:
+		toReturn.Query = &v2_org.SearchQuery_IdQuery{
+			IdQuery: &v2_org.OrganizationIDQuery{
+				Id: assertedType.IdFilter.GetId(),
+			},
+		}
+	case *v2beta_org.OrganizationSearchFilter_NameFilter:
+		toReturn.Query = &v2_org.SearchQuery_NameQuery{
+			NameQuery: &v2_org.OrganizationNameQuery{
+				Name:   assertedType.NameFilter.GetName(),
+				Method: TextQueryMethodBetaToV2(assertedType.NameFilter.GetMethod()),
+			},
+		}
+	case *v2beta_org.OrganizationSearchFilter_StateFilter:
+		toReturn.Query = &v2_org.SearchQuery_StateQuery{
+			StateQuery: &v2_org.OrganizationStateQuery{
+				State: organizationStateBetaToV2(assertedType.StateFilter.GetState()),
+			},
+		}
+	default:
+		return toReturn
+	}
+	return toReturn
+}
+
+func organizationStateBetaToV2(in v2beta_org.OrgState) v2_org.OrganizationState {
+	switch in {
+	case v2beta_org.OrgState_ORG_STATE_ACTIVE:
+		return v2_org.OrganizationState_ORGANIZATION_STATE_ACTIVE
+	case v2beta_org.OrgState_ORG_STATE_INACTIVE:
+		return v2_org.OrganizationState_ORGANIZATION_STATE_INACTIVE
+	case v2beta_org.OrgState_ORG_STATE_REMOVED:
+		return v2_org.OrganizationState_ORGANIZATION_STATE_REMOVED
+	case v2beta_org.OrgState_ORG_STATE_UNSPECIFIED:
+		fallthrough
+	default:
+		return v2_org.OrganizationState_ORGANIZATION_STATE_UNSPECIFIED
+	}
+}
+
+/*
+ * Domain Model to GRPC v2
+ */
+
+func DomainOrganizationListModelToGRPCResponse(orgs []*domain.Organization) []*v2_org.Organization {
+	toReturn := make([]*v2_org.Organization, len(orgs))
+
+	for i, org := range orgs {
+		toReturn[i] = domainOrganizationModelToGRPCResponse(org)
+	}
+
+	return toReturn
+}
+
+func domainOrganizationModelToGRPCResponse(org *domain.Organization) *v2_org.Organization {
+	return &v2_org.Organization{
+		Id: org.ID,
+		Details: &object.Details{
+			ChangeDate:   timestamppb.New(org.UpdatedAt),
+			CreationDate: timestamppb.New(org.CreatedAt),
+		},
+		State:         v2_org.OrganizationState(org.State),
+		Name:          org.Name,
+		PrimaryDomain: org.PrimaryDomain(),
+	}
+}
+
+/*
+ * Domain Model to GRPC v2 beta
+ */
+
+// TODO(IAM-Marco): Remove in V5 (see https://github.com/zitadel/zitadel/issues/10877)
+func DomainOrganizationListModelToGRPCBetaResponse(orgs []*domain.Organization) []*v2beta_org.Organization {
+	toReturn := make([]*v2beta_org.Organization, len(orgs))
+
+	for i, org := range orgs {
+		toReturn[i] = domainOrganizationModelToGRPCBetaResponse(org)
+	}
+
+	return toReturn
+}
+
+// TODO(IAM-Marco): Remove in V5 (see https://github.com/zitadel/zitadel/issues/10877)
```

Comment:
I would add the TODO before the first line because we can remove the whole file afterwards